pyfr/plugins/kernels/turbulence.mako

```diff
@@ -0,0 +1,65 @@
+<% import numpy as np %>
+
+<%namespace module='pyfr.backends.base.makoutil' name='pyfr'/>
+
+<%pyfr:macro name='turbulence' params='t, u, ploc, src'>
+  fpdtype_t tbc = 5.9604644775390625e-8;
+  fpdtype_t arg;
+  fpdtype_t clip;
+  fpdtype_t g;
+  fpdtype_t tpos[3];
+  fpdtype_t tploc[3];
+  fpdtype_t delta2[3];
+  fpdtype_t yzmin[2] = {${ymin}, ${zmin}};
+  fpdtype_t yzmax[2] = {${ymax}, ${zmax}};
+  fpdtype_t utilde[3] = {};
+
+  unsigned int oldstate;
+  unsigned int newstate;
+  unsigned int rshift;
+  
+  % for i, r in enumerate(rot):
+    tploc[${i}] = ${' + '.join(f'{r[j]}*(ploc[{j}] - {shift[j]})' for j in range(3))};
+  % endfor
+
+  % for i in range(nvmax):
+    tpos[0] = ${-ls} + (t-tinit[${i}][0])*${ubar};
+
+    % for j in range(2):
+      % if loop.index == 0:
+        oldstate = state[${i}][0];
+      % else:
+        oldstate = newstate;
+      % endif
+      newstate = (oldstate * 747796405UL) + 2891336453UL;
+      rshift = oldstate >> (28UL);
+      oldstate ^= oldstate >> (4UL + rshift);
+      oldstate *= 277803737UL;
+      oldstate ^= oldstate >> 22UL;
+      tpos[${j+1}] = yzmin[${j}] + (yzmax[${j}]-yzmin[${j}])*((fpdtype_t)(oldstate >> 8UL) * tbc);
+    % endfor
+
+    arg = 0.0;
+    % for j in range(3):
+      delta2[${j}] = (tpos[${j}] - tploc[${j}])*(tpos[${j}] - tploc[${j}]);
+      arg += ${fac1}*delta2[${j}];
```

Comment:
Remove fac1 and put into polyfit.

Reply:
fac1 can be e.g. order 100. When I put fac1 in the polyfit, polyfit has to fit a poly to ~ e^(100x) (albeit in a restricted domain of x) - this results in very large +- poly coefficients and instability/crash. My proposal is to just use polyfit for exp(x) and then feed the poly scaled arguments.

Reply:
That line of reasoning really does not make a lot of sense.  As for the fit to work it has to include points of ~100*x (since that is what will be passed in when we actually call the function).